tests/v2/integration/catalogv2/system_charts_test.go

```diff
@@ -0,0 +1,439 @@
+	w.Require().NoError(err)
+	clusterRepo.Spec.GitRepo = "https://github.com/rancher/charts-small-fork"
+	clusterRepo.Spec.GitBranch = "aks-integration-test-2"
+	clusterRepo, err = w.catalogClient.ClusterRepos().Update(ctx, clusterRepo, metav1.UpdateOptions{})
+	w.Require().NoError(err)
+
+	origClusterRepo := clusterRepo.DeepCopy()
+	downloadTime := clusterRepo.Status.DownloadTime
+	w.Require().NoError(w.pollUntilDownloaded("rancher-charts", downloadTime))
+	cfg, err := w.client.Steve.SteveType(configmaps.ConfigMapSteveType).ByID(fmt.Sprintf("%s/%s", origClusterRepo.Status.IndexConfigMapNamespace, origClusterRepo.Status.IndexConfigMapName))
+	w.Require().NoError(err)
+	cfgMap := &v1.ConfigMap{}
+	w.Require().NoError(stevev1.ConvertToK8sType(cfg.JSONResp, cfgMap))
+	origCfg := cfgMap.DeepCopy()
+
+	// GETTING INDEX FROM CONFIGMAP AND MODIFYING iT
+	gz, err := gzip.NewReader(bytes.NewBuffer(cfgMap.BinaryData["content"]))
+	w.Require().NoError(err)
+	defer gz.Close()
+	data, err := io.ReadAll(gz)
+	w.Require().NoError(err)
+	index := &repo.IndexFile{}
+	w.Require().NoError(json.Unmarshal(data, index))
+	index.SortEntries()
+	index.Entries["rancher-aks-operator"] = index.Entries["rancher-aks-operator"][len(index.Entries["rancher-aks-operator"])-1:]
+	//index.Entries["rancher-aks-operator-crd"] = index.Entries["rancher-aks-operator-crd"][len(index.Entries["rancher-aks-operator"])-1:]
+	marshal, err := json.Marshal(index)
+	w.Require().NoError(err)
+	var compressedData bytes.Buffer
+	writer := gzip.NewWriter(&compressedData)
+	_, err = writer.Write(marshal)
+	w.Require().NoError(err)
+	w.Require().NoError(writer.Close())
+	cfgMap.BinaryData["content"] = compressedData.Bytes()
+
+	//UPDATING THE CONFIGMAP
+	cfg, err = w.client.Steve.SteveType(configmaps.ConfigMapSteveType).Update(cfg, cfgMap)
+	w.Require().NoError(stevev1.ConvertToK8sType(cfg.JSONResp, cfgMap))
+
+	//Updating the cluster
+	w.Require().NoError(w.updateManagementCluster())
+
+	app, err := w.waitForAksChart(rv1.StatusDeployed, "rancher-aks-operator", 0)
+	w.Require().NoError(err)
+
+	ops := w.catalogClient.Operations("cattle-system")
+	list, err := ops.List(ctx, metav1.ListOptions{})
+	w.Require().NoError(err)
+	numberOfOps := countNumberOfOperations(list, "rancher-aks-operator")
+
+	//REVERT CONFIGMAP TO ORIGINAL VALUE
+	cfgMap.BinaryData["content"] = origCfg.BinaryData["content"]
+	cfg, err = w.client.Steve.SteveType(configmaps.ConfigMapSteveType).Update(cfg, cfgMap)
+	w.Require().NoError(err)
+
+	clusterRepo, err = w.catalogClient.ClusterRepos().Get(ctx, "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	clusterRepo.Spec.ForceUpdate = &metav1.Time{Time: time.Now()}
+	_, err = w.catalogClient.ClusterRepos().Update(context.TODO(), clusterRepo.DeepCopy(), metav1.UpdateOptions{})
+	w.Require().NoError(err)
+
+	_, err = w.waitForAksChart(rv1.StatusFailed, "rancher-aks-operator", app.Spec.Version)
+	w.Require().NoError(err)
+	list, err = ops.List(ctx, metav1.ListOptions{})
```

Comment:
should we add a sleep here ? and then check for countNumberofoperations to check for if it is still only 1 operation pod. I think sleep of 2 3 minutes should be good since in the original bug operation pods were getting created every 10 seconds